controllers/challenge_controller_test.go

```diff
@@ -0,0 +1,223 @@
+			context5 := helpers.CreateTestContext("", `{
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 20
+			}`)
+			challengeController.Create(context5)
+			Expect(context5.Writer.Status()).To(Equal(http.StatusBadRequest))
+
+		}))
+
+		It("Test No Location Info", helpers.TestWithMongo("ChallengeController-Create-No-Location", func() {
+
+			// Attempt creating a challenge with only the minimum required fields.
+			context := helpers.CreateTestContext("", `{
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 20
+			}`)
+			challengeController.Create(context)
+			Expect(context.Writer.Status()).To(Equal(http.StatusOK))
+
+			// Test that the challenge was written to mongo.
+			challenge := &models.Challenge{}
+			_ = mgm.Coll(challenge).First(bson.M{}, challenge)
+			Expect(challenge).NotTo(BeNil())
+
+			// Attempt creating a challenge with the same taskName.
+			duplicateContext := helpers.CreateTestContext("", `{
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 20
+			}`)
+			challengeController.Create(duplicateContext)
+			Expect(duplicateContext.Writer.Status()).To(Equal(http.StatusBadRequest))
+
+		}))
+
+		It("Test Invalid Arguments", helpers.TestWithMongo("ChallengeController-Create-Invalid-Args", func() {
+
+			// Create a sample request for a challenge to be created, with an invalid (negative) point value.
+			context := helpers.CreateTestContext("", `{
+				"latitude": 10.01,
+				"longitude": 10.01,
+				"address": "1234 Main Street",
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": -1
+			}`)
+
+			// Test the challenge, and that the response is bad.
+			challengeController.Create(context)
+			Expect(context.Writer.Status()).To(Equal(http.StatusBadRequest))
+
+			// Create a sample request for a challenge to be created, with an invalid (decimal) point value.
+			context2 := helpers.CreateTestContext("", `{
+				"latitude": 10.01,
+				"longitude": 10.01,
+				"address": "1234 Main Street",
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 10.5
+			}`)
+
+			// Test the challenge, and that the response is bad.
+			challengeController.Create(context2)
+			Expect(context2.Writer.Status()).To(Equal(http.StatusBadRequest))
+
+			// Attempt creating a challenge with invalid lat/long coordinates (out of bounds).
+			context3 := helpers.CreateTestContext("", `{
+				"latitude": 500,
+				"longitude": -500,
+				"address": "1234 Main Street",
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 20
+			}`)
+
+			// Test the challenge, and that the response is bad.
+			challengeController.Create(context3)
+			Expect(context3.Writer.Status()).To(Equal(http.StatusBadRequest))
+
+			// Create a sample request for a challenge to be created, with invalid lat/long values.
+			context4 := helpers.CreateTestContext("", `{
```

Comment:
how does this test something different that context3 doesn't already test?

Reply:
fixed